### ISSUE TITLE
Error compiling data-types.h

### ISSUE THREAD

**Issue by 0323pin** [kovidgoyal/kitty#5135]
**Describe the bug**
Trying to update `kitty` from 0.25.0 to 0.25.1 on NetBSD-current. Fail to compile data-types.h

**To Reproduce**
Compile `kitty`-0.25.1 from source.

```
Compiling kitty/unicode-data.c ...
cc -MMD -DNDEBUG -I/usr/pkg/include -I/usr/X11R7/include/freetype2 -I/usr/include -I/usr/X11R7/include -I/usr/pkg/include/ncurses -I/usr/pkg/include/harfbuzz -I/usr/pkg/include/glib-2.0 -I/usr/pkg/include/gio-unix-2.0 -I/usr/pkg/lib/glib-2.0/include -I/usr/X11R7/include/libdrm -I/usr/pkg/include/python3.9 -DPRIMARY_VERSION=4000 -DSECONDARY_VERSION=25 -DXT_VERSION="0.25.1" -Wextra -Wfloat-conversion -Wno-missing-field-initializers -Wall -Wstrict-prototypes -std=c11 -pedantic-errors -Werror -O3 -fwrapv -fstack-protector-strong -pipe -fvisibility=hidden -D_FORTIFY_SOURCE=2 -fPIC -pipe -O2 -fPIC -D_FORTIFY_SOURCE=2 -I/usr/pkg/include -I/usr/X11R7/include/freetype2 -I/usr/include -I/usr/X11R7/include -I/usr/pkg/include/ncurses -I/usr/pkg/include/harfbuzz -I/usr/pkg/include/glib-2.0 -I/usr/pkg/include/gio-unix-2.0 -I/usr/pkg/lib/glib-2.0/include -I/usr/X11R7/include/libdrm -I/usr/pkg/include/python3.9 -flto -DKITTY_HAS_RS_SIG_ARGS -pthread -I/usr/pkg/include/libpng16 -I/usr/pkg/include -I/usr/X11R7/include -I/usr/pkg/include/harfbuzz -I/usr/X11R7/include/freetype2 -I/usr/pkg/include -I/usr/pkg/include/glib-2.0 -I/usr/pkg/lib/glib-2.0/include -I/usr/pkg/include/python3.9 -c kitty/unicode-data.c -o build/fast_data_types-unicode-data.c.o
In file included from kitty/unicode-data.c:3:
kitty/data-types.h:167:15: error: expected declaration specifiers or '...' before 'sizeof'
167 | static_assert(sizeof(GPUCell) == 20, "Fix the ordering of GPUCell");
| ^~~~~~
kitty/data-types.h:167:38: error: expected declaration specifiers or '...' before string constant
167 | static_assert(sizeof(GPUCell) == 20, "Fix the ordering of GPUCell");
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
kitty/data-types.h:174:15: error: expected declaration specifiers or '...' before 'sizeof'
174 | static_assert(sizeof(CPUCell) == 12, "Fix the ordering of CPUCell");
| ^~~~~~
kitty/data-types.h:174:38: error: expected declaration specifiers or '...' before string constant
174 | static_assert(sizeof(CPUCell) == 12, "Fix the ordering of CPUCell");
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*** Error code 1

Stop.
```

**Environment details**
```
NetBSD-9.99.97
Python-3.9.13
```

**Comment by kovidgoyal** [comment]
Looks like your compiler is too old for static_assert just comment out
those lines till you have a chance to update the compiler.

**Comment by 0323pin** [comment]
> Looks like your compiler is too old for static_assert just comment out those lines till you have a chance to update the compiler.

I can force the package to require a newer version of `gcc`. Hope `gcc`-10 will do.
Thanks

**Comment by kovidgoyal** [comment]
IIRC static_assert was introduced a while back, since gcc 4.3

**Comment by 0323pin** [comment]
> IIRC static_assert was introduced a while back, since gcc 4.3

This is weird then, I need to check the default `gcc` version but, I'm rather sure it is `gcc`-9. Double checking as soon as I'm at my machine again.

**Comment by kovidgoyal** [comment]
There may be some other reason static_assert is not defined in netbsd, I
am not familiar with that OS

**Comment by kovidgoyal** [comment]
static_assert is part of c11 and kitty is compiling with -std=c11

**Comment by 0323pin** [comment]
> static_assert is part of c11 and kitty is compiling with -std=c11

Aha... thanks!
I'll try that, right now we have `USE_LANGUAGES= c99` on the package `Makefile`

**Comment by 0323pin** [comment]
@kovidgoyal problem solved, update to 0.25.1 merged by @0-wiz-0. The issue was `USE_LANGUAGES= c99` in the package `Makefile`.
